runner/src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -341,13 +424,25 @@ export class PageControllerBase {
         ? redirectTo(request, h, startPage)
         : redirectTo(request, h, `/${this.model.basePath}${startPage}`);
       this.setFeedbackDetails(viewModel, request);
+
+      //Starting from the start page and through to this.page, calculate the route we took building up a "routeState" containing just the values we should have populated
+      //Once we've found the route back to the currently requested page we will resolve our calculated state
+      const routeState = await new Promise((resolve) => {
+        //Grab the start page definition
+        const startPageDef = this.model.pages.find(
+          (page) => page.path === startPage
+        );
```

Comment:
Interesting, wasn't aware of this - Can see this also takes into account the condition checks. Thanks @superafroman, will review the PR again and consider taking advantage of this rather than using my own condition checks.

Reply:
Again, no longer applicable as this is already done as part of @superafroman 's proposed solution.